src/groupeddataframe/utils.jl

```diff
@@ -303,6 +306,21 @@ function row_group_slots(cols::NTuple{N, AbstractVector},
     else
         sorted = false
     end
+
+    lg = length(groups)
+    nt = Threads.nthreads()
+    use_threading = nt > 1 && lg > 1_000_000 && ngroups < lg * (0.5 - 1 / (2 * nt)) / (2 * nt)
```

Comment:
This formula should be roughly OK and safe (at least on laptops but this is a worst case):
```
~/Desktop/Dev/DF_dev$ julia --project -e 'using DataFrames, BenchmarkTools; n=100_000_000; df=DataFrame(passband=rand(1:3_000_000,n)); @btime groupby($df,:passband)'
  674.282 ms (71 allocations: 765.80 MiB)
~/Desktop/Dev/DF_dev$ julia --project -t 4 -e 'using DataFrames, BenchmarkTools; n=100_000_000; df=DataFrame(passband=rand(1:3_000_000,n)); @btime groupby($df,:passband)'
  532.848 ms (694 allocations: 774.44 MiB)
~/Desktop/Dev/DF_dev$ julia --project -t 8 -e 'using DataFrames, BenchmarkTools; n=100_000_000; df=DataFrame(passband=rand(1:3_000_000,n)); @btime groupby($df,:passband)'
  675.997 ms (71 allocations: 765.80 MiB)
```

Reply:
indeed in the future we should probably increase chunk size to make sure we optimally balance the number of needed `seen` vectors and reduction size (as e.g. in the above case splitting the job into 2 chunks is better than using 8 threads).

Reply:
I have added a comment in TODO.